web/src/engine/websites/NetComics.ts

```diff
@@ -0,0 +1,119 @@
+import { Choice } from '../SettingsManager';
+import { EngineResourceKey as E, WebsiteResourceKey as W, TagResourceKey as R } from '../../i18n/ILocale';
+import * as Common from './decorators/Common';
+import { FetchJSON } from '../platform/FetchProvider';
+
+type APIResult<T> = {
+    status: number,
+    message: string,
+    data: T
+}
+
+type APIManga = {
+    title_id: number,
+    site: string,
+    title_name: string,
+}
+
+type APIChapter = {
+    chapter_id: number,
+    chapter_no: number,
+    chapter_name: string
+}
+
+type APIPages = {
+    images: {
+        image_url: string
+    }[]
+}
+
+@Common.ImageAjax()
+
+export default class extends DecoratableMangaScraper {
+    private readonly apiUrl = 'https://beta-api.netcomics.com';
+
+    public constructor() {
+        super('netcomics', `NetComics`, 'https://www.netcomics.com', Tags.Language.English, Tags.Language.Spanish, Tags.Language.French, Tags.Language.German, Tags.Media.Manhwa, Tags.Source.Official);
+
+        this.Settings.language = new Choice('mangalist.language',
+            E.Settings_Global_Language, //TODO: create a "Preferred language" string?
+            E.Settings_Global_LanguageInfo, //TODO : use proper langage setting string description: "This will affect manga list or smg like that"
+            'EN',
+            { key: 'EN', label: R.Tags_Language_English }, //TODO : create langage string without flag emoji or support emoji in dialog?
+            { key: 'ES', label: R.Tags_Language_Spanish },
+            { key: 'FR', label: R.Tags_Language_French },
+            { key: 'DE', label: R.Tags_Language_German },
+        );
+
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return /https:\/\/www\.netcomics\.com\/[a-z]{2}\/comic\/[^/]+/.test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const site = url.match(/\/([a-z]{2})\/comic\//)[1].toUpperCase();
+        const slug = url.match(/\/comic\/(\S+)/)[1];
+        const request = new Request(new URL(`/api/v1/title/comic/${slug}/${site}`, this.apiUrl), {
+            headers: this.getRequestHeaders(site)
+        });
+        const { data } = await FetchJSON<APIResult<APIManga>>(request);
+        return new Manga(this, provider, data.title_id.toString(), data.title_name.trim());
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList : Manga[]= [];
+        for (let page = 1, run = true; run; page++) {
+            const mangas = await this.getMangasFromPage(page, provider);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList.distinct();
+    }
+
+    async getMangasFromPage(page: number, provider: MangaPlugin) {
+        const request = new Request(new URL(`/api/v1/title/genre?no=${page}&size=18&genre=`, this.apiUrl), {
```

Comment:
How about fetching mangas for all languages?

Reply:
Hum, why not